hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoop/HadoopPlugin.groovy

```diff
@@ -55,12 +57,64 @@ class HadoopPlugin implements Plugin<Project> {
   }
 
   /**
+   * Create hadoop configurations - hadoopRuntime and provided
+   * @param project The Gradle project
+   */
+  void addHadoopConfigurations(Project project) {
+    addHadoopRuntimeConfiguration(project);
+    addProvidedConfiguration(project);
+  }
+
+  /**
+   * Prepare the "provided" Hadoop configuration for the project.
+   *
+   * @param project The Gradle project
+   * @return The provided configuration
+   */
+  Configuration addProvidedConfiguration(Project project) {
+
+    // create provided configuration and set it as a private configuration.
+    Configuration provided = project.getConfigurations().create("provided");
+    provided.setVisible(false);
+
+    // we get default, testCompile, compile and runtime configurations from java plugin.
+    project.getPlugins().withType(JavaPlugin) {
+
+      // remove the provided dependencies from default configuration.
+      provided.dependencies.all {
+        dep -> project.getConfigurations().getByName('default').exclude group: dep.group, module: dep.name
+      }
+
+      // testCompile should extend from provided so that all the provided dependencies including transitive
+      // dependencies are on the classpath to facilitate testing.
+      project.getConfigurations().getByName("testCompile").extendsFrom(provided);
+
+      // add provided jars on the classpath, this will not add transitive dependencies of each jar.
+      project.sourceSets.main.compileClasspath += provided
+      project.javadoc.classpath += provided
+
+      // Idea doesn't understand the above classpath modification. Adding it explicitly.
+      project.getPlugins().withType(IdeaPlugin) {
+        project.idea.module.scopes.PROVIDED.plus += [provided]
+      }
+    }
+
+    // Eclipse understands the classpath, but add it, to be on the safe side.
+    project.getPlugins().withType(EclipsePlugin) {
```

Comment:
Should this be inside the project.getPlugins().withType(JavaPlugin) { } code block?